src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,28 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("Can't add country " + country, e);
+        } finally {
+            session.close();
```

Comment:
Potential resource leak: session should be closed in a finally block to ensure it is closed even if an exception is thrown. Use try-with-resources or try-finally construct.